test/common/upstream/health_checker_impl_test.cc

```diff
@@ -4150,7 +4380,8 @@ TEST(HealthCheckProto, Validation) {
     unhealthy_threshold: 1
     no_traffic_interval: 0s
     http_health_check:
-      service_name: locations
+      service_name_matcher:
```

Comment:
We still need to test the old behavior. Can you make sure there is still a test that covers the deprecated field? There are two things here:
1) parseHealthCheckFromV2Yaml does not correctly schema validate. Please make this use loadFromYamlAndValidate so it will correctly check for deprecated fields.
2) You should have some DEPRECATED_FEATURE_TEST that covers the old behavior.